connectors/src/connectors/confluence/index.ts

```diff
@@ -403,3 +380,71 @@ export async function retrieveConfluenceObjectsTitles(
 
   return new Ok(titles);
 }
+
+export async function retrieveConfluenceResourceParents(
+  connectorId: ModelId,
+  internalId: string
+): Promise<Result<string[], Error>> {
+  const confluenceId = getIdFromConfluenceInternalId(internalId);
+
+  if (isConfluenceInternalPageId(internalId)) {
+    const currentPage = await ConfluencePage.findOne({
+      attributes: ["pageId", "parentId", "spaceId"],
+      where: {
+        connectorId,
+        pageId: confluenceId,
+      },
+    });
+
+    if (!currentPage) {
+      return new Err(new Error("Confluence page not found."));
+    }
+
+    // If the page does not have a parentId, return only the spaceId.
+    if (!currentPage.parentId) {
+      return new Ok([makeConfluenceInternalSpaceId(currentPage.spaceId)]);
+    }
+
+    // Currently opting for a best-effort strategy to reduce database queries,
+    // this logic may be enhanced later for important Confluence connections.
+    // By fetching all pages within a space, we reconstruct parent-child
+    // relationships in-app, minimizing database interactions.
+    // If needed we could move the same approach as Notion and cache the results in Redis.
```

Comment:
Notion caches in redis because we use it during ingestion more than because of this route AFAICT.

Independently seems a bit bullish to fetch all the pages at once. But fine with it. Let's keep an eye on speed in the assitsant builder resource selection display :+1:

Reply:
Also I though confluence was flat (spaces only are shown to the user. In which case we don't need that implementation).

This is:
- used in the resource picker for building an assistant retrieval to show where things have been selected in the tree before expanding the tree
- has to be aligned with getConnectorResourcePermission. If it's flat there, then this should imply return []

Reply:
Based on the implementation of `retrieveConfluenceConnectorPermissions` I believe we are flat and you can simply return [] no?

Reply:
Seen IRL, when the view permission is set to read, our goal is to present the entire hierarchy to demonstrate to users the data that is accessible within Dust. However, when users are in the section for selecting new data to import, we will display only Spaces.

I'll add some comments and improve naming in the code to make this more explicit.

Reply:
Updated.